test/e2e/antreapolicy_test.go

```diff
@@ -1705,6 +1705,11 @@ func TestAntreaPolicy(t *testing.T) {
 		t.Run("Case=ACNPAllowXBtoA", func(t *testing.T) { testACNPAllowXBtoA(t) })
 		t.Run("Case=ACNPAllowXBtoYA", func(t *testing.T) { testACNPAllowXBtoYA(t) })
 		t.Run("Case=ACNPPriorityOverrideDefaultDeny", func(t *testing.T) { testACNPPriorityOverrideDefaultDeny(t) })
+		// The test group cleans up ACNP/ANPs before executing a test case. However, cleanupDefaultDenyNPs() validates
+		// that all Pod to Pod traffic is allowed without cleaning up Antrea-native policies, which fails if there is
+		// a lingering policy from a previous test run. Clean up all Antrea-native policies before exiting the group.
+		failOnError(k8sUtils.CleanACNPs(), t)
+		failOnError(k8sUtils.CleanANPs(namespaces), t)
```

Comment:
this seems out of place: why are the individual test cases (e.g. `testACNPAllowXBtoA`) not responsible for cleaning-up the individual Antrea-native policies they create?

Reply:
When I extended the old netpol test framework to Antrea-native policy, the testcase execution logic was: before each testcase is run, cleanup any netpol that are in the cluster. So essentially any resource created by a testcase is cleaned-up by the subsequent testcase. I simply re-used this logic in antreapolicy_test but I think it has caused enough confusion and issues by now. I agree that maybe we should refactor the execute testcase logic.

Reply:
maybe we should, but in a different PR ? i would like to unblock a few PRs failing on this behavior.. unless we are releasing next week in which case i will change the behavior in this PR

Reply:
still failing.. so that's not the root cause :(

Reply:
@abhiraut your e2e tests are failing for a different reason... there's a hard 35m timeout for Kind tests, and since you added more CG related testcases, you will always exceed that limit. You need to increase it in `ci/kind/test-e2e-kind.sh`. The issue you raised however is valid and maybe I'll fix it by changing the test execution logic.

Reply:
let me retry with new value